homeassistant/helpers/entity.py

```diff
@@ -223,7 +223,10 @@ def convert_to_entity_category(
                 "EntityCategory instead" % (type(value).__name__, value),
                 error_if_core=False,
             )
-        return EntityCategory(value)
+        try:
+            return EntityCategory(value)
+        except ValueError:
+            return None
```

Comment:
Should we log something here?

Reply:
It is already logged above that it wasn't valid in the first place :)

Reply:
`raise_report` argument is set to `False` in `async_load`, because we are reading from json and it is always a string

https://github.com/home-assistant/core/blob/768de8d515ca002eb45e3982a1785a626efdafe8/homeassistant/helpers/entity_registry.py#L631-L633

Reply:
It will get logged when set by the integration causing it.

Reply:
But only when the integration is writing to it - not when reading from the json.
This will never inform the user if it is already in the registry.

Reply:
If it's already there, it's too late.
And the logging done here does not help at all...

```
Stack (most recent call last):
2022-02-11 15:59:48 WARNING (MainThread) [homeassistant.helpers.frame] Detected code that uses str (invalid) for entity category. This is deprecated and will stop working in Home Assistant 2022.4, it should be updated to use EntityCategory instead. Please report this issue.
Stack (most recent call last):
  File "/usr/local/bin/hass", line 33, in <module>
    sys.exit(load_entry_point('homeassistant', 'console_scripts', 'hass')())
  File "/workspaces/core/homeassistant/__main__.py", line 191, in main
    exit_code = runner.run(runtime_conf)
  File "/workspaces/core/homeassistant/runner.py", line 119, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
    handle._run()
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 510, in _async_add_entity
    entry = entity_registry.async_get_or_create(
  File "/workspaces/core/homeassistant/helpers/entity_registry.py", line 353, in async_get_or_create
    return self.async_update_entity(
  File "/workspaces/core/homeassistant/helpers/entity_registry.py", line 565, in async_update_entity
    new = self.entities[entity_id] = attr.evolve(old, **new_values)
  File "/usr/local/lib/python3.9/site-packages/attr/_funcs.py", line 343, in evolve
    return cls(**changes)
  File "<attrs generated init homeassistant.helpers.entity_registry.RegistryEntry>", line 13, in __init__
    _setattr('entity_category', __attr_converter_entity_category(entity_category))
  File "/workspaces/core/homeassistant/helpers/entity_registry.py", line 103, in _convert_to_entity_category
    return convert_to_entity_category(value, raise_report=raise_report)
  File "/workspaces/core/homeassistant/helpers/entity.py", line 220, in convert_to_entity_category
    report(
  File "/workspaces/core/homeassistant/helpers/frame.py", line 74, in report
    _LOGGER.warning(msg, stack_info=True)
```

"informing" the user about this is pretty useless anyway.